service/controller/resource/ipam/azure_machinepool_persister.go

```diff
@@ -0,0 +1,117 @@
+package ipam
+
+import (
+	"context"
+	"net"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	"sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/pkg/annotation"
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+type AzureMachinePoolPersisterConfig struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+type AzureMachinePoolPersister struct {
+	ctrlClient client.Client
+	logger     micrologger.Logger
+}
+
+func NewAzureMachinePoolPersister(config AzureMachinePoolPersisterConfig) (*AzureMachinePoolPersister, error) {
+	if config.CtrlClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.G8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	p := &AzureMachinePoolPersister{
+		ctrlClient: config.CtrlClient,
+		logger:     config.Logger,
+	}
+
+	return p, nil
+}
+
+func (p *AzureMachinePoolPersister) Persist(ctx context.Context, vnet net.IPNet, obj interface{}) error {
```

Comment:
We save the allocated subnet:
- as annotation in the `AzureMachinPool`.
- in the `AzureCluster.Spec.NetworkSpec.Subnets` . This will make the `AzureCluster` controller to actually create the subnet.